src/lib/blkalloc/append_blk_allocator.h

```diff
@@ -84,7 +78,7 @@ class AppendBlkAllocator : public BlkAllocator {
     BlkAllocStatus alloc(blk_count_t nblks, blk_alloc_hints const& hints, BlkId& out_blkid) override;
     void free(BlkId const& b) override;
     BlkAllocStatus alloc_on_disk(BlkId const& in_bid) override;
-    BlkAllocStatus mark_blk_allocated(BlkId const& b) override;
+    BlkAllocStatus alloc_on_cache(BlkId const& b) override;
```

Comment:
It looks like with this PR, alloc_on_cache and alloc is the same.

Reply:
alloc_on_cache is only during recovery, because alloc() is done during regular operation, which maintains information in the cache, but during recovery, whatever was committed needs to be put back in cache, at that time, alloc_on_cache() is called internally by homestore.